reflect.go

```diff
@@ -257,6 +256,27 @@ func (r *Reflector) reflectTypeToSchema(definitions Definitions, t reflect.Type)
 		}
 
 	case reflect.Map:
+		switch t.Key().Kind() {
+		case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+			rt := &Type{
+				Type: "object",
+				PatternProperties: map[string]*Type{
+					"^[0-9]+$": r.reflectTypeToSchema(definitions, t.Elem()),
+				},
+				AdditionalProperties: []byte("false"),
+			}
+			return rt
+		case reflect.Float32, reflect.Float64:
+			rt := &Type{
+				Type: "object",
+				PatternProperties: map[string]*Type{
+					"^[-+]?[0-9]*\\.?[0-9]+$": r.reflectTypeToSchema(definitions, t.Elem()),
```

Comment:
Unfortunately a float regex is much more complex than this, and hard to get right. For example this doesn't support exponents.

 It might be simpler just to not worry about floats.

Reply:
This one would: ^[-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?$

What do you think ?